### PR TITLE
hamster additions

### PR DESCRIPTION
**Pull request by shimonewman**

### CODE REVIEW

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 1]
```diff
@@ -277,7 +277,7 @@ private function getStatsForProject(Document $project, Group $pools, Cache $cach
                         $limit = $periodValue['limit'];
                         $period = $periodValue['period'];
 
-                        $requestDocs = $dbForProject->find('stats_v2', [
+                        $requestDocs = $dbForProject->find('stats', [
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:280]: lets revert this back to stats_v2

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 2]
```diff
@@ -368,6 +368,19 @@ private function getStatsForOrganization(Document $organization, Database $dbFor
                 throw new \Exception('Membership not found. Skipping organization : ' . $organization->getId());
             }
 
+            $billingPlan = $membership->getAttribute('billingPlan', null);
+            $billingPlanDowngrade = $membership->getAttribute('billingPlanDowngrade', null);
+
+            if (!empty($billingPlan) && empty($billingPlanDowngrade)) {
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:374]:
```suggestion
            if (!empty($billingPlan) && is_null($billingPlanDowngrade)) {
```

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 3]
```diff
@@ -368,6 +368,19 @@ private function getStatsForOrganization(Document $organization, Database $dbFor
...
+                $statsPerOrganization['billing-plan'] = $billingPlan;
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:375]:
```suggestion
                $statsPerOrganization['billing_plan'] = $billingPlan;
```

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 4]
```diff
@@ -368,6 +368,19 @@ private function getStatsForOrganization(Document $organization, Database $dbFor
...
+            }
+
+            if (in_array($billingPlan, ['tier-1', 'tier-2'])) {
+                $billingStartDate = $membership->getAttribute('billingStartDate', null);
+                $statsPerOrganization['billing-start-plan-date'] = $billingStartDate;
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:380]:
```suggestion
                $statsPerOrganization['billing_start_date'] = $billingStartDate;
```

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 5]
```diff
@@ -410,6 +423,24 @@ protected function getStatsPerUser(Document $user, Database $dbForConsole)
...
+            $membership = $dbForConsole->findOne('teams', [
+                Query::equal('userInternalId', [$user->getInternalId()])
+            ]);
+
+            if (!empty($membership) || !$membership->isEmpty()) {
+                $billingPlan = $membership->getAttribute('billingPlan', null);
+                $billingPlanDowngrade = $membership->getAttribute('billingPlanDowngrade', null);
+
+                if (!empty($billingPlan) && empty($billingPlanDowngrade)) {
+                    $statsPerUser['billing-plan'] = $billingPlan;
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:435]:
```suggestion
                    $statsPerUser['billing_plan'] = $billingPlan;
```

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 6]
```diff
@@ -410,6 +423,24 @@ protected function getStatsPerUser(Document $user, Database $dbForConsole)
...
+                }
+
+                if (in_array($billingPlan, ['tier-1', 'tier-2'])) {
+                    $billingStartDate = $membership->getAttribute('billingStartDate', null);
+                    $statsPerUser['billing-start-plan-date'] = $billingStartDate;
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:440]:
```suggestion
                    $statsPerUser['billing_start_date'] = $billingStartDate;
```

File: src/Appwrite/Platform/Workers/Hamster.php (hunk 1 above)
Comment (shimonewman) [reply src/Appwrite/Platform/Workers/Hamster.php:280]: sure

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 7]
```diff
@@ -368,6 +367,21 @@ private function getStatsForOrganization(Document $organization, Database $dbFor
                 throw new \Exception('Membership not found. Skipping organization : ' . $organization->getId());
             }
 
+            $billingPlan = $membership->getAttribute('billingPlan', null);
+            $billingPlanDowngrade = $membership->getAttribute('billingPlanDowngrade', null);
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:371]: This is the membership collection.. membership collection does not have these attributes. these attributes are part of the teams collection

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 8]
```diff
@@ -410,6 +424,24 @@ protected function getStatsPerUser(Document $user, Database $dbForConsole)
         try {
             $statsPerUser = [];
 
+            $membership = $dbForConsole->findOne('teams', [
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:427]:
```suggestion
            $team = $dbForConsole->findOne('teams', [
```

File: src/Appwrite/Platform/Workers/Hamster.php [hunk 9]
```diff
@@ -410,6 +424,24 @@ protected function getStatsPerUser(Document $user, Database $dbForConsole)
...
+                Query::equal('userInternalId', [$user->getInternalId()])
+            ]);
+
+            if (!empty($membership) || !$membership->isEmpty()) {
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Hamster.php:431]:
```suggestion
            if (!empty($membership) && !$membership->isEmpty()) {
```

File: src/Appwrite/Platform/Workers/Hamster.php (hunk 8 above)
Comment (shimonewman) [reply src/Appwrite/Platform/Workers/Hamster.php]: i use $organization for consistency

File: src/Appwrite/Platform/Workers/Hamster.php (hunk 7 above)
Comment (shimonewman) [reply src/Appwrite/Platform/Workers/Hamster.php]: Yeah, it should be $organization.
fixed

File: src/Appwrite/Platform/Workers/Hamster.php (hunk 9 above)
Comment (shimonewman) [reply src/Appwrite/Platform/Workers/Hamster.php]: fixed

**Pull request merged**